### ISSUE TITLE
Suddenly RStudio treats Rmd yaml as R code

### ISSUE THREAD

**Issue by RosieRedfield** [STAT545-UBC/Discussion#365]
This has happened twice now.

I had a Rmd file that was working fine for the markdown side of things - each time I ran it the R chunks were executed to the extent that their coding errors allowed, and the markdown instructions text was ignored.  I could also knit it with no markdown problems; the header 'yaml' information was correctly applied.

For no reason I can discern, the next time I ran it I got error messages throughout, indicating that R was now trying to interpret the header yaml info ( title, author, date and output) as R commands.  This continued to happen each time I tried to run it, not only with the file I was editing but with a previously saved Rmd file that had been working fine.    Quitting and restarting RStudio didn't solve the problem.  Copying the header into a new Rmd file didn't solve the problem.

This first happened a couple of days ago.  I eventually gave up trying to figure out what was wrong.  Something I did this morning (?) created a file that works, but now it's happened again.

Suggestions?

**Comment by jennybc** [comment]
I know that I have had some real tragedies when "reflowing" an Rmd document. We have discussed reflowing comments before, yes? Well, if you "reflow" an Rmd document but you are not in an R comment, I think it can reflow too much and mangle your YAML. Perhaps this has happened?

**Comment by RosieRedfield** [comment]
I haven't been using the reflow comments command, and messing up the YAML in one file wouldn't explain why a different file that had previously been fine now shows the same problem.  

I think this morning's 'solution' might have been that I opened the file on a different computer than the one I'd been using while travelling.  Could it be that, while trying to type a keyboard shortcut, I accidentally hit a keyboard combination that changed some deep-buried RStudio setting that prevents it from interpreting Rmd documents?  This would explain why the problem now appears in previously-OK files.  I've been through the RStudio list of shortcuts but none look like the likely explanation.  I'll keep searching.

**Comment by jennybc** [comment]
No, what you say doesn't ring any bells or resonate with my spidey sense. I'm not sure what's going on. Hopefully you can show one of these situations to me or a TA.

**Comment by RosieRedfield** [comment]
Could it be a bug in this new preview version of RStudio?

**Comment by jennybc** [comment]
It is possible. But my experience has taught me that I am usually the bug. So I consider that unlikely.

**Comment by samhinshaw** [comment]
Hi @RosieRedfield 

YAML in R Markdown documents is incredibly finicky, and can be thrown off entirely by one wrong keystroke!  My most common errors with YAML headers in the past are:

1. The YAML header is not on the **VERY first line** of the R Markdown document
2. The YAML header is corrupted somehow, for example:
```yaml
 ---
title: "My RMarkdown Document"
output: github_document
---
```
Did you catch the error there? Just one space off, and those lines are no longer treated as a header!

Try playing spot the difference between the working YAML and the non-working YAML R Markdown docs. I know it sounds silly, but a surprising amount of my errors are due to fatfingering keys.
